app/lib/csv_importer.rb

```diff
@@ -1,41 +1,56 @@
-module CsvImporter
-  CSV_CATEGORIES = {
-      SPAWNING_SUCCESS: 'Spawning Success',
-      TAGGED_ANIMAL_ASSESSMENT: 'Tagged Animal Assessment',
-      UNTAGGED_ANIMAL_ASSESSMENT: 'Untagged Animal Assessment'
-  }.freeze
+class CsvImporter
+  attr_reader :stats, :model, :filename, :processed_file_id
 
-  class InvalidCsvCategoryError < StandardError; end;
+  CATEGORIES = [
+      'Spawning Success',
+      'Tagged Animal Assessment',
+      'Untagged Animal Assessment',
+      'Wild Collection',
+  ].freeze
+
+  class InvalidCategoryError < StandardError; end;
 
   def self.import(filename, category_name, processed_file_id)
-    csv_category_model = model_from_category(category_name)
-    IOStreams.each_record(filename) do |record|
-      attrs = translate_attribute_names(record, category_name)
-      attrs[:processed_file_id] = processed_file_id
-      csv_row_record = csv_category_model.new(attrs.merge({raw: false}))
-      csv_row_record.cleanse_data! if csv_row_record.respond_to?(:cleanse_data!)
-      csv_row_record.save
+    importer = new(filename, category_name, processed_file_id)
+    importer.process
+    importer.stats
+  end
+
+  def initialize(filename, category_name, processed_file_id)
+    @filename = filename
+    @processed_file_id = processed_file_id
+    @model = model_from_category(category_name)
+
+    @stats = Hash.new(0)
+    @stats[:shl_case_numbers] = Hash.new(0)
+  end
+
+  def process
+    IOStreams.each_record(filename) do |csv_row|
+      csv_row[:processed_file_id] = processed_file_id
+      csv_row[:raw] = false
+      record = model.create_from_csv_data(csv_row)
+      record.cleanse_data! if record.respond_to?(:cleanse_data!)
```

Comment:
It's probable that whatever data cleansing is happening here can move to inside the `create_from_csv_data()` methods.

Reply:
If the data cleansing is minimal, I think it's fine. But if it's some heavy business logic, would the model object be the best place to put it?